### ISSUE TITLE
Enhanced System Observability for Multi-Server Environments (Unified Endpoints?)

### ISSUE THREAD

**Issue by dezoito** [ollama/ollama#8247]
As Ollama adoption grows, the lack of comprehensive system metrics makes it challenging to meet standard operational requirements - monitoring, alerting, and planning across development, staging, and production environments.

This can also prevent a wider adoption in commercial and production applications.

While the current endpoints (`/api/version`, `/api/tags`, `/api/ps`) provide basic information, consolidating and expanding these into a single observability endpoint would significantly improve monitoring and management capabilities, 

Proposed: New `/api/info` endpoint returning unified system metrics, like:

```json
{
  "version": "0.1.16",
  "system": {
    "cpu": {
      "cores": 16,
      "threads": 32,
      "usage_percent": 45.2,
      ...
    },
    "memory": {
      "total_bytes": 34359738368,
      "used_bytes": 28859738368,
      ...
    },
    "gpus": [
      {
        "name": "NVIDIA GeForce RTX 4090",
        "memory": {
          "total_bytes": 25769803776,
          "used_bytes": 16106127360,
          ...
        },
      }
    ]
  },
  "models": {
    "loaded": [...],
    "available": [...],
  },
}
```

This enhancement would:

- Enable proper monitoring and alerting in production environments
- Simplify capacity planning across deployments
- Allow integration with standard observability tools

To the community:

- What other metrics or stats would be useful to have?
- How do you currently monitor or observe your running Ollama instances?

**Comment by maglore9900** [comment]
Agree. This would be very helpful for enterprise type deployments, but also helpful for power users, and potentially troubleshooting.

**Comment by kth8** [comment]
Prometheus and Grafana is wildly used in commercial and production and the standard endpoint for that is `/metrics`. In order for Prometheus to scrape that endpoint, the format should be like [this](https://demo.promlabs.com/metrics) instead of json.

**Comment by dezoito** [comment]
> In order for Prometheus to scrape that endpoint, the format should be like [this](https://demo.promlabs.com/metrics) instead of json.

Thank you for the feedback!

That looks way more detailed, of course.

I still think a simple JSON response could still add a ton of value (and would be easier to implement by combining the existing endpoints).

Someone cue in the "why not both" meme, please...

**Comment by rick-github** [comment]
https://github.com/ollama/ollama/issues/3144

**Comment by rick-github** [comment]
closing as dupe #3144
